# Tips Dataset

## Definition

The main reason for collecting data for Tips dataset was to analyse the factors influencing the food server's tips in restaurant. The collection of data was conducted in one restaurant, located in a suburban shopping centre during an interval of two and a half months in early 1990. The following categories of data were assembled on all food served customers:

* Total bill (cost of the meal)including tax, in US dollars
* Tip in US dollars
* Sex of the person paying for the meal
* Is smoker in party?
* Day of the week
* Time of the day
* Size of the party


To import various libraries as numpy, seaborn (plotting), pandas

In [26]:
import numpy as np

In [27]:
import seaborn as sns

In [28]:
import pandas as pd

To upload the file from the url address and create the dataframe using Pandas

In [29]:
df=pd.read_csv("https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/tips.csv")

In [6]:
df

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
5         25.29  4.71    Male     No   Sun  Dinner     4
6          8.77  2.00    Male     No   Sun  Dinner     2
7         26.88  3.12    Male     No   Sun  Dinner     4
8         15.04  1.96    Male     No   Sun  Dinner     2
9         14.78  3.23    Male     No   Sun  Dinner     2
10        10.27  1.71    Male     No   Sun  Dinner     2
11        35.26  5.00  Female     No   Sun  Dinner     4
12        15.42  1.57    Male     No   Sun  Dinner     2
13        18.43  3.00    Male     No   Sun  Dinner     4
14        14.83  3.02  Female     No   Sun  Dinner     2
15        21.58  3.92    Male     No   Sun  Dinner     2
16        10.33  1.67  Female     No   Sun  Dinner     3
17        16.29  3.71    Male     No   Sun  Dinner     3
18        16.97  3.50  Female     No   Sun  Dinner     3
19        20.65  3.35    Male     No   Sat  Dinner     3
20        17.92  4.08    Male     No   Sat  Dinner     2
21        20.29  2.75  Female     No   Sat  Dinner     2
22        15.77  2.23  Female     No   Sat  Dinner     2
23        39.42  7.58    Male     No   Sat  Dinner     4
24        19.82  3.18    Male     No   Sat  Dinner     2
25        17.81  2.34    Male     No   Sat  Dinner     4
26        13.37  2.00    Male     No   Sat  Dinner     2
27        12.69  2.00    Male     No   Sat  Dinner     2
28        21.70  4.30    Male     No   Sat  Dinner     2
29        19.65  3.00  Female     No   Sat  Dinner     2
..          ...   ...     ...    ...   ...     ...   ...
214       28.17  6.50  Female    Yes   Sat  Dinner     3
215       12.90  1.10  Female    Yes   Sat  Dinner     2
216       28.15  3.00    Male    Yes   Sat  Dinner     5
217       11.59  1.50    Male    Yes   Sat  Dinner     2
218        7.74  1.44    Male    Yes   Sat  Dinner     2
219       30.14  3.09  Female    Yes   Sat  Dinner     4
220       12.16  2.20    Male    Yes   Fri   Lunch     2
221       13.42  3.48  Female    Yes   Fri   Lunch     2
222        8.58  1.92    Male    Yes   Fri   Lunch     1
223       15.98  3.00  Female     No   Fri   Lunch     3
224       13.42  1.58    Male    Yes   Fri   Lunch     2
225       16.27  2.50  Female    Yes   Fri   Lunch     2
226       10.09  2.00  Female    Yes   Fri   Lunch     2
227       20.45  3.00    Male     No   Sat  Dinner     4
228       13.28  2.72    Male     No   Sat  Dinner     2
229       22.12  2.88  Female    Yes   Sat  Dinner     2
230       24.01  2.00    Male    Yes   Sat  Dinner     4
231       15.69  3.00    Male    Yes   Sat  Dinner     3
232       11.61  3.39    Male     No   Sat  Dinner     2
233       10.77  1.47    Male     No   Sat  Dinner     2
234       15.53  3.00    Male    Yes   Sat  Dinner     2
235       10.07  1.25    Male     No   Sat  Dinner     2
236       12.60  1.00    Male    Yes   Sat  Dinner     2
237       32.83  1.17    Male    Yes   Sat  Dinner     2
238       35.83  4.67  Female     No   Sat  Dinner     3
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

_Dataset has 244 rows and 7 columns. By checking the last 5 lines we can see if  all data are showing correctly:_

In [7]:
df.tail()


total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

_To inspect the type of data in the columns._

In [8]:
df.dtypes

total_bill    float64
tip           float64
sex            object
smoker         object
day            object
time           object
size            int64
dtype: object

_The basic statistics can be called by __. describe()__ function:_

In [9]:
df.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

The describe() function provides the calculation of mean, standard deviation, minimal, maximal value and median (50%). The sample contains 244 records of different food served customers about their sex, size of the group, meal type and day when they were served, how much was their bill and how much they pay the tip.

In average USD19.79 was spent in total and average USD1.38 of tips was paid. 

In [78]:
pd.pivot_table(df, index=["time","sex"],aggfunc="size")

time    sex   
Dinner  Female     52
        Male      124
Lunch   Female     35
        Male       33
dtype: int64

In [60]:
pd.pivot_table(df, index=["time","sex"],aggfunc=[np.mean, np.max, np.min])

mean                       amax                     \
                   size       tip total_bill   day size smoker    tip   
time   sex                                                              
Dinner Female  2.461538  3.002115  19.213077  Thur    5    Yes   6.50   
       Male    2.701613  3.144839  21.461452   Sun    6    Yes  10.00   
Lunch  Female  2.457143  2.582857  16.339143  Thur    6    Yes   5.17   
       Male    2.363636  2.882121  18.048485  Thur    6    Yes   6.70   

                         amin                               
              total_bill  day size smoker   tip total_bill  
time   sex                                                  
Dinner Female      44.30  Fri    1     No  1.00       3.07  
       Male        50.81  Fri    2     No  1.00       7.25  
Lunch  Female      43.11  Fri    1     No  1.25       8.35  
       Male        41.19  Fri    1     No  1.44       7.51

In [61]:
pd.pivot_table(df, index=["day"],aggfunc=[np.mean])

mean                     
          size       tip total_bill
day                                
Fri   2.105263  2.734737  17.151579
Sat   2.517241  2.993103  20.441379
Sun   2.842105  3.255132  21.410000
Thur  2.451613  2.771452  17.682742

In [11]:
df["smoker"].describe()

count     244
unique      2
top        No
freq      151
Name: smoker, dtype: object

In [12]:
smoker=df[df.loc[:,"smoker"]=="Yes"]
                

In [13]:
smoker


total_bill   tip     sex smoker   day    time  size
56        38.01  3.00    Male    Yes   Sat  Dinner     4
58        11.24  1.76    Male    Yes   Sat  Dinner     2
60        20.29  3.21    Male    Yes   Sat  Dinner     2
61        13.81  2.00    Male    Yes   Sat  Dinner     2
62        11.02  1.98    Male    Yes   Sat  Dinner     2
63        18.29  3.76    Male    Yes   Sat  Dinner     4
67         3.07  1.00  Female    Yes   Sat  Dinner     1
69        15.01  2.09    Male    Yes   Sat  Dinner     2
72        26.86  3.14  Female    Yes   Sat  Dinner     2
73        25.28  5.00  Female    Yes   Sat  Dinner     2
76        17.92  3.08    Male    Yes   Sat  Dinner     2
80        19.44  3.00    Male    Yes  Thur   Lunch     2
83        32.68  5.00    Male    Yes  Thur   Lunch     2
90        28.97  3.00    Male    Yes   Fri  Dinner     2
92         5.75  1.00  Female    Yes   Fri  Dinner     2
93        16.32  4.30  Female    Yes   Fri  Dinner     2
95        40.17  4.73    Male    Yes   Fri  Dinner     4
96        27.28  4.00    Male    Yes   Fri  Dinner     2
97        12.03  1.50    Male    Yes   Fri  Dinner     2
98        21.01  3.00    Male    Yes   Fri  Dinner     2
100       11.35  2.50  Female    Yes   Fri  Dinner     2
101       15.38  3.00  Female    Yes   Fri  Dinner     2
102       44.30  2.50  Female    Yes   Sat  Dinner     3
103       22.42  3.48  Female    Yes   Sat  Dinner     2
105       15.36  1.64    Male    Yes   Sat  Dinner     2
106       20.49  4.06    Male    Yes   Sat  Dinner     2
107       25.21  4.29    Male    Yes   Sat  Dinner     2
109       14.31  4.00  Female    Yes   Sat  Dinner     2
138       16.00  2.00    Male    Yes  Thur   Lunch     2
164       17.51  3.00  Female    Yes   Sun  Dinner     2
..          ...   ...     ...    ...   ...     ...   ...
203       16.40  2.50  Female    Yes  Thur   Lunch     2
204       20.53  4.00    Male    Yes  Thur   Lunch     4
205       16.47  3.23  Female    Yes  Thur   Lunch     3
206       26.59  3.41    Male    Yes   Sat  Dinner     3
207       38.73  3.00    Male    Yes   Sat  Dinner     4
208       24.27  2.03    Male    Yes   Sat  Dinner     2
209       12.76  2.23  Female    Yes   Sat  Dinner     2
210       30.06  2.00    Male    Yes   Sat  Dinner     3
211       25.89  5.16    Male    Yes   Sat  Dinner     4
213       13.27  2.50  Female    Yes   Sat  Dinner     2
214       28.17  6.50  Female    Yes   Sat  Dinner     3
215       12.90  1.10  Female    Yes   Sat  Dinner     2
216       28.15  3.00    Male    Yes   Sat  Dinner     5
217       11.59  1.50    Male    Yes   Sat  Dinner     2
218        7.74  1.44    Male    Yes   Sat  Dinner     2
219       30.14  3.09  Female    Yes   Sat  Dinner     4
220       12.16  2.20    Male    Yes   Fri   Lunch     2
221       13.42  3.48  Female    Yes   Fri   Lunch     2
222        8.58  1.92    Male    Yes   Fri   Lunch     1
224       13.42  1.58    Male    Yes   Fri   Lunch     2
225       16.27  2.50  Female    Yes   Fri   Lunch     2
226       10.09  2.00  Female    Yes   Fri   Lunch     2
229       22.12  2.88  Female    Yes   Sat  Dinner     2
230       24.01  2.00    Male    Yes   Sat  Dinner     4
231       15.69  3.00    Male    Yes   Sat  Dinner     3
234       15.53  3.00    Male    Yes   Sat  Dinner     2
236       12.60  1.00    Male    Yes   Sat  Dinner     2
237       32.83  1.17    Male    Yes   Sat  Dinner     2
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2

[93 rows x 7 columns]

In [14]:
nosmok=df[df.loc[:,"smoker"]=="No"]

In [15]:
nosmok

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
5         25.29  4.71    Male     No   Sun  Dinner     4
6          8.77  2.00    Male     No   Sun  Dinner     2
7         26.88  3.12    Male     No   Sun  Dinner     4
8         15.04  1.96    Male     No   Sun  Dinner     2
9         14.78  3.23    Male     No   Sun  Dinner     2
10        10.27  1.71    Male     No   Sun  Dinner     2
11        35.26  5.00  Female     No   Sun  Dinner     4
12        15.42  1.57    Male     No   Sun  Dinner     2
13        18.43  3.00    Male     No   Sun  Dinner     4
14        14.83  3.02  Female     No   Sun  Dinner     2
15        21.58  3.92    Male     No   Sun  Dinner     2
16        10.33  1.67  Female     No   Sun  Dinner     3
17        16.29  3.71    Male     No   Sun  Dinner     3
18        16.97  3.50  Female     No   Sun  Dinner     3
19        20.65  3.35    Male     No   Sat  Dinner     3
20        17.92  4.08    Male     No   Sat  Dinner     2
21        20.29  2.75  Female     No   Sat  Dinner     2
22        15.77  2.23  Female     No   Sat  Dinner     2
23        39.42  7.58    Male     No   Sat  Dinner     4
24        19.82  3.18    Male     No   Sat  Dinner     2
25        17.81  2.34    Male     No   Sat  Dinner     4
26        13.37  2.00    Male     No   Sat  Dinner     2
27        12.69  2.00    Male     No   Sat  Dinner     2
28        21.70  4.30    Male     No   Sat  Dinner     2
29        19.65  3.00  Female     No   Sat  Dinner     2
..          ...   ...     ...    ...   ...     ...   ...
150       14.07  2.50    Male     No   Sun  Dinner     2
151       13.13  2.00    Male     No   Sun  Dinner     2
152       17.26  2.74    Male     No   Sun  Dinner     3
153       24.55  2.00    Male     No   Sun  Dinner     4
154       19.77  2.00    Male     No   Sun  Dinner     4
155       29.85  5.14  Female     No   Sun  Dinner     5
156       48.17  5.00    Male     No   Sun  Dinner     6
157       25.00  3.75  Female     No   Sun  Dinner     4
158       13.39  2.61  Female     No   Sun  Dinner     2
159       16.49  2.00    Male     No   Sun  Dinner     4
160       21.50  3.50    Male     No   Sun  Dinner     4
161       12.66  2.50    Male     No   Sun  Dinner     2
162       16.21  2.00  Female     No   Sun  Dinner     3
163       13.81  2.00    Male     No   Sun  Dinner     2
165       24.52  3.48    Male     No   Sun  Dinner     3
166       20.76  2.24    Male     No   Sun  Dinner     2
167       31.71  4.50    Male     No   Sun  Dinner     4
185       20.69  5.00    Male     No   Sun  Dinner     5
195        7.56  1.44    Male     No  Thur   Lunch     2
212       48.33  9.00    Male     No   Sat  Dinner     4
223       15.98  3.00  Female     No   Fri   Lunch     3
227       20.45  3.00    Male     No   Sat  Dinner     4
228       13.28  2.72    Male     No   Sat  Dinner     2
232       11.61  3.39    Male     No   Sat  Dinner     2
233       10.77  1.47    Male     No   Sat  Dinner     2
235       10.07  1.25    Male     No   Sat  Dinner     2
238       35.83  4.67  Female     No   Sat  Dinner     3
239       29.03  5.92    Male     No   Sat  Dinner     3
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[151 rows x 7 columns]

In [16]:
pd.pivot_table(smoker,index=["time","day"],values=["tip","total_bill"], aggfunc=[np.mean, np.min,np.max,len])

mean             amin              amax              len  \
                  tip total_bill   tip total_bill    tip total_bill   tip   
time   day                                                                  
Dinner Fri   3.003333  19.806667  1.00       5.75   4.73      40.17   9.0   
       Sat   2.875476  21.276667  1.00       3.07  10.00      50.81  42.0   
       Sun   3.516842  24.120000  1.50       7.25   6.50      45.35  19.0   
Lunch  Fri   2.280000  12.323333  1.58       8.58   3.48      16.27   6.0   
       Thur  3.030000  19.190588  2.00      10.34   5.00      43.11  17.0   

                        
            total_bill  
time   day              
Dinner Fri         9.0  
       Sat        42.0  
       Sun        19.0  
Lunch  Fri         6.0  
       Thur       17.0

In [17]:
pd.pivot_table(nosmok,index=["time","day"],values=["tip","total_bill","size"], aggfunc=[np.mean, np.min,np.max,len])

mean                      amin                  amax       \
                 size       tip total_bill size   tip total_bill size  tip   
time   day                                                                   
Dinner Fri   2.000000  2.750000  19.233333    2  1.50      12.46    2  3.5   
       Sat   2.555556  3.102889  19.661778    1  1.00       7.25    4  9.0   
       Sun   2.929825  3.167895  20.506667    2  1.01       8.77    6  6.0   
       Thur  2.000000  3.000000  18.780000    2  3.00      18.78    2  3.0   
Lunch  Fri   3.000000  3.000000  15.980000    3  3.00      15.98    3  3.0   
       Thur  2.500000  2.666364  17.075227    1  1.25       7.51    6  6.7   

                        len                   
            total_bill size   tip total_bill  
time   day                                    
Dinner Fri       22.75    3   3.0        3.0  
       Sat       48.33   45  45.0       45.0  
       Sun       48.17   57  57.0       57.0  
       Thur      18.78    1   1.0        1.0  
Lunch  Fri       15.98    1   1.0        1.0  
       Thur      41.19   44  44.0       44.0

In [18]:
df["day"].describe()

count     244
unique      4
top       Sat
freq       87
Name: day, dtype: object

In [75]:
pd.pivot_table(df, index=["day"],aggfunc="size")

day
Fri     19
Sat     87
Sun     76
Thur    62
dtype: int64

In [77]:
pd.pivot_table(df, index=["day"],aggfunc=[np.mean, np.min,np.max])

mean                         amin                            \
          size       tip total_bill     sex size smoker    time   tip   
day                                                                     
Fri   2.105263  2.734737  17.151579  Female    1     No  Dinner  1.00   
Sat   2.517241  2.993103  20.441379  Female    1     No  Dinner  1.00   
Sun   2.842105  3.255132  21.410000  Female    2     No  Dinner  1.01   
Thur  2.451613  2.771452  17.682742  Female    1     No  Dinner  1.25   

                 amax                                        
     total_bill   sex size smoker    time    tip total_bill  
day                                                          
Fri        5.75  Male    4    Yes   Lunch   4.73      40.17  
Sat        3.07  Male    5    Yes  Dinner  10.00      50.81  
Sun        7.25  Male    6    Yes  Dinner   6.50      48.17  
Thur       7.51  Male    6    Yes   Lunch   6.70      43.11

In [19]:
df[df.loc[:,"day"]=="Sat"].describe()

total_bill        tip       size
count   87.000000  87.000000  87.000000
mean    20.441379   2.993103   2.517241
std      9.480419   1.631014   0.819275
min      3.070000   1.000000   1.000000
25%     13.905000   2.000000   2.000000
50%     18.240000   2.750000   2.000000
75%     24.740000   3.370000   3.000000
max     50.810000  10.000000   5.000000

In [20]:
df["time"].describe()

count        244
unique         2
top       Dinner
freq         176
Name: time, dtype: object

In [21]:
df["sex"].describe()

count      244
unique       2
top       Male
freq       157
Name: sex, dtype: object

In [22]:
df[df.loc[:,"sex"]=="Female"].describe()

total_bill        tip       size
count   87.000000  87.000000  87.000000
mean    18.056897   2.833448   2.459770
std      8.009209   1.159495   0.937644
min      3.070000   1.000000   1.000000
25%     12.750000   2.000000   2.000000
50%     16.400000   2.750000   2.000000
75%     21.520000   3.500000   3.000000
max     44.300000   6.500000   6.000000

To get better understanding about different variables, the % representation of gender, days when tips paid and the preferred time. 

The dataset named "Female" created which represents all varaibles just for female. The results on the bottom shows that 87 out of 244 tips were paid by females. 

In [23]:
female=df[df.loc[:,"sex"]=="Female"]


In [24]:
female

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
11        35.26  5.00  Female     No   Sun  Dinner     4
14        14.83  3.02  Female     No   Sun  Dinner     2
16        10.33  1.67  Female     No   Sun  Dinner     3
18        16.97  3.50  Female     No   Sun  Dinner     3
21        20.29  2.75  Female     No   Sat  Dinner     2
22        15.77  2.23  Female     No   Sat  Dinner     2
29        19.65  3.00  Female     No   Sat  Dinner     2
32        15.06  3.00  Female     No   Sat  Dinner     2
33        20.69  2.45  Female     No   Sat  Dinner     4
37        16.93  3.07  Female     No   Sat  Dinner     3
51        10.29  2.60  Female     No   Sun  Dinner     2
52        34.81  5.20  Female     No   Sun  Dinner     4
57        26.41  1.50  Female     No   Sat  Dinner     2
66        16.45  2.47  Female     No   Sat  Dinner     2
67         3.07  1.00  Female    Yes   Sat  Dinner     1
71        17.07  3.00  Female     No   Sat  Dinner     3
72        26.86  3.14  Female    Yes   Sat  Dinner     2
73        25.28  5.00  Female    Yes   Sat  Dinner     2
74        14.73  2.20  Female     No   Sat  Dinner     2
82        10.07  1.83  Female     No  Thur   Lunch     1
85        34.83  5.17  Female     No  Thur   Lunch     4
92         5.75  1.00  Female    Yes   Fri  Dinner     2
93        16.32  4.30  Female    Yes   Fri  Dinner     2
94        22.75  3.25  Female     No   Fri  Dinner     2
100       11.35  2.50  Female    Yes   Fri  Dinner     2
101       15.38  3.00  Female    Yes   Fri  Dinner     2
102       44.30  2.50  Female    Yes   Sat  Dinner     3
103       22.42  3.48  Female    Yes   Sat  Dinner     2
..          ...   ...     ...    ...   ...     ...   ...
155       29.85  5.14  Female     No   Sun  Dinner     5
157       25.00  3.75  Female     No   Sun  Dinner     4
158       13.39  2.61  Female     No   Sun  Dinner     2
162       16.21  2.00  Female     No   Sun  Dinner     3
164       17.51  3.00  Female    Yes   Sun  Dinner     2
168       10.59  1.61  Female    Yes   Sat  Dinner     2
169       10.63  2.00  Female    Yes   Sat  Dinner     2
178        9.60  4.00  Female    Yes   Sun  Dinner     2
186       20.90  3.50  Female    Yes   Sun  Dinner     3
188       18.15  3.50  Female    Yes   Sun  Dinner     3
191       19.81  4.19  Female    Yes  Thur   Lunch     2
197       43.11  5.00  Female    Yes  Thur   Lunch     4
198       13.00  2.00  Female    Yes  Thur   Lunch     2
201       12.74  2.01  Female    Yes  Thur   Lunch     2
202       13.00  2.00  Female    Yes  Thur   Lunch     2
203       16.40  2.50  Female    Yes  Thur   Lunch     2
205       16.47  3.23  Female    Yes  Thur   Lunch     3
209       12.76  2.23  Female    Yes   Sat  Dinner     2
213       13.27  2.50  Female    Yes   Sat  Dinner     2
214       28.17  6.50  Female    Yes   Sat  Dinner     3
215       12.90  1.10  Female    Yes   Sat  Dinner     2
219       30.14  3.09  Female    Yes   Sat  Dinner     4
221       13.42  3.48  Female    Yes   Fri   Lunch     2
223       15.98  3.00  Female     No   Fri   Lunch     3
225       16.27  2.50  Female    Yes   Fri   Lunch     2
226       10.09  2.00  Female    Yes   Fri   Lunch     2
229       22.12  2.88  Female    Yes   Sat  Dinner     2
238       35.83  4.67  Female     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[87 rows x 7 columns]

In [25]:
pd.pivot_table(female,values="tip",index=["time","day"], columns=["sex"])

sex            Female
time   day           
Dinner Fri   2.810000
       Sat   2.801786
       Sun   3.367222
       Thur  3.000000
Lunch  Fri   2.745000
       Thur  2.561935